ngraph_bridge/ie_executable.cc

```diff
@@ -141,5 +155,33 @@ bool IE_Executable::call(const vector<shared_ptr<runtime::Tensor>>& outputs,
   m_infer_req.Infer();
   return true;
 }
+
+bool IE_Executable::call_trivial(
+    const vector<shared_ptr<runtime::Tensor>>& outputs,
+    const vector<shared_ptr<runtime::Tensor>>& inputs) {
+  // Do basic sanity checking first
+  if (outputs.size() != 1) {
+    THROW_IE_EXCEPTION << "Expected only one output for trivial functions";
+  }
+  if (!inputs.size() && !m_hoisted_params.size()) {
+    THROW_IE_EXCEPTION
+        << "Called trivial IE function with no inputs or hoisted params";
+  }
+  if (inputs.size() && inputs.size() != 1) {
+    THROW_IE_EXCEPTION << "Called trivial IE function with multiple inputs";
+  }
+  if (m_hoisted_params.size() && m_hoisted_params.size() != 1) {
+    THROW_IE_EXCEPTION
+        << "Called trivial IE function with multiple hoisted params";
+  }
+
+  auto input = m_hoisted_params.size() ? m_hoisted_params[0].second : inputs[0];
+  auto size = input->get_size_in_bytes();
+  unsigned char* buf_ptr = new unsigned char[size];
```

Comment:
Alloc/Dealloc in every iteration will slow down inference. Can we optimize with saving a map of name->buffer, along with sanity-check size every time and possibly some heuristics to purge LRU cache if we cross a threshold.

Reply:
Um, what? We could eliminate the temporary but the input and output tensors could be allocated on different devices.

Reply:
Yes my comment was about the temp buf

Reply:
Ah, so some form of buffer caching? I think it's probably an overkill to consider it at this moment for short-circuiting trivial functions.

Reply:
Ok for now, as our target models won't probably have such Const->Result disjoint subgraphs in a ng-func. In GATK type model we saw plenty of these patterns. We can monitor performance and take action later.